kernel/sched.c

```diff
@@ -397,15 +399,43 @@ static void move_thread_to_end_of_prio_q(struct k_thread *thread)
 	update_cache(thread == _current);
 }
 
-static void flag_ipi(void)
+static void flag_ipi(uint32_t ipi_mask)
 {
 #if defined(CONFIG_SMP) && defined(CONFIG_SCHED_IPI_SUPPORTED)
 	if (arch_num_cpus() > 1) {
-		_kernel.pending_ipi = true;
+		atomic_or(&_kernel.pending_ipi, (atomic_val_t)ipi_mask);
 	}
 #endif
 }
 
+/* Create a bitmask of CPUs that need an IPI. Note: sched_spinlock is held. */
+static atomic_val_t ipi_mask_create(struct k_thread *thread)
+{
+#if defined(CONFIG_SMP) && (CONFIG_MP_MAX_NUM_CPUS > 1)
+	uint32_t  ipi_mask = 0;
+	struct k_thread *cpu_thread;
+	unsigned int num_cpus = arch_num_cpus();
+	uint32_t  id = arch_proc_id();
+
+	for (uint32_t i = 0; i < num_cpus; i++) {
```

Comment:
Nit: inconsistent mix of `unsigned` and `uint32_t` to number CPUs.

Reply:
Fixed.